mangaki/mangaki/views.py

```diff
@@ -60,6 +63,119 @@
 GHIBLI_IDS = [2591, 8153, 2461, 53, 958, 30, 1563, 410, 60, 3315, 3177, 106]
 
 
+def current_user_ratings(request, works=None):
+    """
+    Compute the set of ratings for the current user.
+
+    This handles both the case where the user is logged in and the case where
+    it is an anonymous user, in which case the rating is fetched from the
+    session.
+
+    If the `works` argument is given, then only the ratings for the given works
+    will be considered.
+
+    Arguments:
+        request -- The Request object we are currently handling.
+        works   -- An iterable of Work instances or primary keys, or None.
+
+    Returns:
+        ratings -- A dictionary mapping Work primary keys to their rating
+            string ('like', 'dislike', etc.)
+    """
+    user = request.user
+    if user.is_anonymous:
+        ratings = request.session.get(settings.ANONYMOUS_RATINGS_SESSION_KEY, {})
+        if works is not None:
+            filtered_ratings = {}
+            for work in works:
+                # Accept Work models as well as primary key integers
+                work = str(getattr(work, 'pk', work))
```

Comment:
Extracted into a `pk_from_object_or_pk` method in the following patch in the series (#278).